### PR TITLE
solvents interface code

### PR DESCRIPTION
**Pull request by mshuaibii**

### CODE REVIEW

File: ocdata/core/complex_solvent_config.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+            metadata = {
+                "adsorbates": adsorbate_metadata,
+                "solvent": self.solvent.name,
+                "ions": [x.name for x in self.ions],
+                "ion_concentrations": [
+                    x.get_ion_concentration(volume) for x in self.ions
+                ],
+                "solvent_depth": self.solvent_depth,
+                "solvent_volume": volume,
```
Comment (mshuaibii) [inline ocdata/core/complex_solvent_config.py:160]: Can we think of other metadata we would like to store when we generate the input? Is initial surface charge density something meaningful (i.e. `ion charge / Surface area`)?

cc - @nitishg91

Comment (nitishg91) [reply ocdata/core/complex_solvent_config.py:160]: Yes, initial surface charge would be good to store.

File: ocdata/core/complex_solvent_config.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+                    f.write(f"structure {tmp_dir}/ion_{i}.pdb\n")
+                    f.write("  number 1\n")
+                    f.write("  fixed 0 0 0 0 0 0\n")
+                    f.write("end structure\n\n")
+
+            self.run_packmol(packmol_input)
+
+            solvent_ions_atoms = ase.io.read(output_path, format="proteindatabank")
+            solvent_ions_atoms.set_pbc(True)
+            solvent_ions_atoms.set_tags([3] * len(solvent_ions_atoms))
```
Comment (levineds) [inline ocdata/core/complex_solvent_config.py:207]: Where does this "3" come from?

File: ocdata/core/complex_solvent_config.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+        try:
+            ps = subprocess.Popen(
+                f"{packmol_cmd} < {packmol_input}",
+                shell=True,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+            out, err = ps.communicate()
+        except NotImplementedError:
+            raise OSError("packmol is not found.")
```
Comment (levineds) [inline ocdata/core/complex_solvent_config.py:223]: Shouldn't this be a test on the output of `which`. As in,
```
if packmol_cmd is None:
    raise OSError('packmol is not found')
```
You're not really trying to catch a `subprocess` error.

**Review by levineds — COMMENTED** [review]
LGTM, units seem right

File: ocdata/core/complex_solvent_config.py (hunk 2 above)
Comment (mshuaibii) [reply ocdata/core/complex_solvent_config.py:207]: This is arbitrary. OC20 methodology used 0 for subsurface atoms, 1 for surface atoms, and 2 for adsorbates. Here I'm using 3 for the solvent environment we're introducing.

Comment (levineds) [reply ocdata/core/complex_solvent_config.py:207]: Ah, a comment to that effect would be useful.

**Review by nitishg91 — COMMENTED** [review]
Everything looks good.

**Comment by mshuaibii** [comment]
Moved into https://github.com/FAIR-Chem/fairchem

**Pull request closed without merging**